### PR TITLE
Adding audience to fluid static

### PR DESCRIPTION
**Pull request by micahgodbolt**
CRA `AppWithAudience.tsx` is just to demo the usage of audience. Not necessarily going to check that in, or check it in as is.

### CODE REVIEW

File: experimental/framework/fluid-static/src/FluidStatic.ts [hunk 1]
```diff
@@ -70,6 +75,12 @@ export class FluidContainer {
         return this.rootDataObject.createDataObject<T>(dataObjectClass, id);
     }
 
+    public getId() {
```
Comment (skylerjokiel) [inline experimental/framework/fluid-static/src/FluidStatic.ts:78]: why not just clientId?

File: experimental/framework/fluid-static/src/FluidStatic.ts [hunk 2]
```diff
@@ -70,6 +75,12 @@ export class FluidContainer {
...
+        return this.container.clientId;
+    }
+
+    public readonly audience: IAudience;
```
Comment (skylerjokiel) [inline experimental/framework/fluid-static/src/FluidStatic.ts:82]: Let's keep public variables at the top of the class. Above constructor below private variables.

File: experimental/framework/fluid-static/src/FluidStatic.ts [hunk 3]
```diff
@@ -39,22 +41,25 @@ export interface ContainerCreateConfig extends ContainerConfig {
     initialDataObjects?: IdToDataObjectCollection;
 }
 
-export class FluidContainer {
+export class FluidContainer extends EventEmitter {
```
Comment (skylerjokiel) [inline experimental/framework/fluid-static/src/FluidStatic.ts:44]: class doesn't need to extend EventEmitter

File: experimental/framework/fluid-static/src/FluidStatic.ts (hunk 1 above)
Comment (micahgodbolt) [reply experimental/framework/fluid-static/src/FluidStatic.ts:78]: are we expecting the user to understand what the 'client' is? or do you think that term is generic enough?

File: experimental/framework/fluid-static/src/FluidStatic.ts (hunk 2 above)
Comment (micahgodbolt) [reply experimental/framework/fluid-static/src/FluidStatic.ts:82]: ok, so just functions down below

File: experimental/framework/fluid-static/src/FluidStatic.ts [hunk 4]
```diff
@@ -39,22 +40,27 @@ export interface ContainerCreateConfig extends ContainerConfig {
...
     private readonly types: Set<string>;
+
+    public readonly container: Container;
+
     public constructor(
-        container: IContainer, // we anticipate using this later, e.g. for Audience
+        readonly incomingContainer: Container, // we anticipate using this later, e.g. for Audience
```
Comment (skylerjokiel) [inline experimental/framework/fluid-static/src/FluidStatic.ts:49]: as a future fyi you can do 

```typescript
public readonly container: Container;
```

File: experimental/framework/fluid-static/src/FluidStatic.ts [hunk 5]
```diff
@@ -39,22 +40,27 @@ export interface ContainerCreateConfig extends ContainerConfig {
     initialDataObjects?: IdToDataObjectCollection;
 }
 
-export class FluidContainer {
+export class FluidContainer extends EventEmitter {
     private readonly types: Set<string>;
+
+    public readonly container: Container;
```
Comment (skylerjokiel) [inline experimental/framework/fluid-static/src/FluidStatic.ts:46]: As we discussed. Lets no do this.

File: examples/apps/cra-demo/src/AppWithAudience.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+    const setupListeners = () => {
+        const updateData = () => setData(dataObject?.query());
+        updateData();
+        dataObject?.on("changed", updateData);
+        container?.container.audience.on("addMember", updateData)
+        container?.container.audience.on("removeMember", updateData)
+        return () => {
+            dataObject?.off("change", updateData)
+            container?.container.audience.off("addMember", updateData)
```
Comment (skylerjokiel) [inline examples/apps/cra-demo/src/AppWithAudience.tsx:66]: This cra change doesn't work anymore.

File: experimental/framework/fluid-static/src/FluidStatic.ts [hunk 7]
```diff
@@ -39,13 +40,19 @@ export interface ContainerCreateConfig extends ContainerConfig {
...
+export class FluidContainer extends EventEmitter implements Pick<Container, "audience" | "clientId"> {
     private readonly types: Set<string>;
+
+    private readonly container: Container;
+
+    public readonly audience: Container["audience"];
+
     public constructor(
-        container: IContainer, // we anticipate using this later, e.g. for Audience
+        container: Container, // we anticipate using this later, e.g. for Audience
```
Comment (skylerjokiel) [inline experimental/framework/fluid-static/src/FluidStatic.ts:51]: can you just declare this inline here and remove the container above

File: examples/apps/cra-demo/src/AppWithAudience.tsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+            setupListeners();
+        }
+    }, [dataObject]);
+
+
+    if (!dataObject || !container) return <div />;
+
+
+    const MemberList = () => {
+        const members: string[] = Array.from(container.audience.getMembers().keys());
```
Comment (skylerjokiel) [inline examples/apps/cra-demo/src/AppWithAudience.tsx:84]: Are you checking in this example?

Also, how does your member list update if you don't listen for new members?

**Review by skylerjokiel — APPROVED** [review]
Let's get this in and iterate. Left some comments around the cra demo if you plan on checking that in.

File: examples/apps/cra-demo/src/AppWithAudience.tsx (hunk 8 above)
Comment (micahgodbolt) [reply examples/apps/cra-demo/src/AppWithAudience.tsx:84]: i had listeners at one point, i really don't need them with this example in that the users name only shows up if they type into the input, at which point means new data and a re-render...Probably a lot more to do to make this example more complete

**Pull request merged**